src/shared_utilities/models.jl

```diff
@@ -383,6 +385,26 @@ end
 
 Domains.coordinates(model::AbstractModel) = Domains.coordinates(model.domain)
 
+"""
+    add_drivers_to_cache(p::NamedTuple, model::AbstractModel, coords)
+
+"""
+function add_drivers_to_cache(p::NamedTuple, model::AbstractModel, coords)
+    (atmos, radiation) = get_drivers(model)
+    vars = initialize_drivers(atmos, radiation, coords)
+    return merge(p, vars)
```

Comment:
might be better to use `merge!` here and update `p` in place?

Reply:
Im not totally sure, but `merge!` is not defined for NamedTuples: https://docs.julialang.org/en/v1/base/collections/#Base.merge!
but,  dicts and named tuples seem closely related - collections of key-value pairs? - but unless we start working with Dicts instead of named tuples in `initialize` and only return the NT at the very end, i am not sure if we can update in place.